core/src/autogluon/core/pseudolabeling/pseudolabeling.py

```diff
@@ -0,0 +1,54 @@
+import numpy as np
+import pandas as pd
+from autogluon.core.constants import PROBLEM_TYPES_CLASSIFICATION
+
+
+def filter_pseudo(y_pred_proba_og, problem_type,
+                  min_proportion_prob: float = 0.05, max_proportion_prob: float = 0.6,
+                  threshold: float = 0.95, percent_sample: float = 0.3):
+    """
+    Takes in the predicted probabilities of the model and chooses the indices that meet
+    a criteria to incorporate into training data. Criteria is determined by problem_type.
+    If mutliclass or binary will choose all rows with max prob over threshold. For regression
+    chooses 30% of the labeled data randomly.
+
+    Parameters:
+    -----------
+    y_pred_proba_og: The predicted probabilities from the current best model. If problem is
+        'binary' or 'multiclass' then it's Panda series of predictive probs, if it's 'regression'
+        then it's a scalar
+    min_proportion_prob: Minimum percentage of total 'y_pred_proba_og` that is below threshold
+        required to trigger threshold change
```

Comment:
This description is unclear to me, could be more clear as something like: 
"Minimum proportion of indices in `y_pred_proba_og` to select. The filter threshold will be automatically adjusted until at least `min_proportion_prob` of the predictions in  `y_pred_proba_og` pass the filter. This ensures we return at least `min_proportion_prob` of the pseudolabeled data to augment the training set in pseudolabeling."